### PR TITLE
Add schema translation hover, completion and MatchingTranslations checking

### PR DESCRIPTION
**Pull request by charlespwd**
## What are you adding in this PR?

- Add `getDefaultSchema{Locale,Translations}(Factory)?` dependencies
- Add `t:` translation completion and hover support in section and theme block `{% schema %}` tags
- Make translation completion _fuzzy_ for a nicer experience
- Add schema translation checking to `MatchingTranslations`
- Add parameter completion for theme translations in Liquid 
    (`{{ count }} items` -> `"path.to.translation" | t: count: count`)

https://github.com/Shopify/theme-tools/assets/4990691/884e8fb0-b048-4b47-a78a-7a8f338ba311

https://github.com/Shopify/theme-tools/assets/4990691/217eb8d0-d1c3-440f-bf82-2418692a7638

## Before you deploy

It's backward compatible, but requires new dependency injections in browser contexts.

- [x] I included a minor bump `changeset`
- [x] I included a patch bump `changeset`

### CODE REVIEW

**Review by karreiro — APPROVED** [review]
Thank you for this PR, @charlespwd! 🔥 I've only left a few minor/non-blocking comments and questions.

My main suggestion is to centralize JSON parsing in the module to keep the rest of the codebase agnostic of the implementation (we could even centralize the `safeParse` there). I'm curious about what you think about this :)

Thanks again for the excellent PR!

File: packages/theme-language-server-common/src/server/startServer.ts [hunk 1]
```diff
@@ -172,7 +188,7 @@ export function startServer(
           codeActionKinds: [...CodeActionKinds],
         },
         completionProvider: {
-          triggerCharacters: ['.', '{{ ', '{% ', '<', '/', '[', '"', "'"],
+          triggerCharacters: ['.', '{{ ', '{% ', '<', '/', '[', '"', "'", ':'],
```
[the remark this answers is not in the document]
Comment (charlespwd) [reply packages/theme-language-server-common/src/server/startServer.ts:191]: There's nothing to complete indeed. The `triggerCharacters` is meant to hint the _client_ to send completion requests when those characters are typed, it doesn't mean that there _should_ be results when they are typed :)

File: packages/theme-check-common/src/test/test-helper.ts [hunk 2]
```diff
@@ -74,9 +75,22 @@ export async function check(
         throw e;
       }
     },
+    async getDefaultSchemaTranslations() {
+      try {
+        return JSON.parse(themeDesc[defaultSchemaTranslationsFileRelativePath] || '{}');
```
[the remark this answers is not in the document]
Comment (charlespwd) [reply packages/theme-check-common/src/test/test-helper.ts:80]: That is probably best, indeed. That said I don't know where I'd put it in the dependency tree. A whole new module just for that seems silly, but an export if theme-check-common also seems silly :p

[the remark this answers is not in the document]
Comment (charlespwd) [reply packages/theme-check-common/src/test/test-helper.ts:80]: I guess we _do_ have `json-to-ast` re-exports in theme-check-common, so maybe not that bad. It's also wrapped in `toSourceCode`...

**Pull request merged**